### PR TITLE
[addons] all user pre-installed addons should be enabled

### PR DESCRIPTION
**Pull request by stefansaraev**
this fixes a regression after e85b8afcc501d48736363480844f705e06d98ba3

@tamland @MilhouseVH

### CODE REVIEW

**Comment by MilhouseVH** [comment]
Thanks for this @stefansaraev, but unfortunately when I delete Addons*.db and start a new OE test build based on Kodi master (03147d78ba3231e001835eb27d99b2da4f659209) and this PR, all system-installed addons in my build (ie. all PVR clients, plus inputstream.mpd) are now enabled and Kodi takes several minutes to start because of one or more PVR clients being slow to start: http://sprunge.us/OYhI

This isn't how Kodi used to behave (ie. last week) when including a bunch of what-should-be-disabled-by-default "system" add-ons in the build (although that wasn't itself ideal, as it would have been nice to have inputstream.mpd enabled by default, but I digress).

Although these PVR addons are installed, since they're part of the build they shouldn't be automatically enabled, and indeed until recently they were not automatically enabled when creating a new Addons2#.db.

Not really sure what the solution is, but perhaps "system" addons that are not included in the `addon-manifest.xml` should not be automatically enabled, and only those "system" addons that are present in the manifest should be auto-enabled? Of course all third-party installed add-ons should be enabled by default.

**Comment by stefansaraev** [comment]
ah, I see, more pvr stupidity. because pvr manager is so f*****g special, and pvr clients were so special (force disabled if not configured but that is gone now - #9429). nothing to do with addonmanager/addondatabase.

pvr manager should start to behave, soon or later..  or is it time to stop shipping pre-installed pvr clients as opdenkamp suggested one year ago ?

however, my point stands. all adons found on first boot should be enabled on first boot (or after deleting addon*db), be it system or user installed. but, see, I did not fight for #9259, I am not going to fight for this. too :)

**Comment by stefansaraev** [comment]
what do our pvr devs think about it. @FernetMenta @ksooo @Jalle19 @xhaggi ?

**Comment by stefansaraev** [comment]
if it is not possible to properly fix pvrmanager, then pvrmanager should support exactly one pvr client addon at a time, configurable via settings -> live tv. just like there can be exactly one skin, audio encoder, visualization or screensaver.

**Comment by FernetMenta** [comment]
I did this "hack" for 15.0 because having had all those pvr and audiodecoder addons sucked: https://github.com/xbmc/xbmc/commit/2f0f01f9a8305bce83ac1047a2e081f109445730

We should stop shipping them and stop building them. The original plan was that we provide the infrastructure for binary addons to be built but not taking ownership for them. Patching more than a dozen pvr addons after an API change is silly. That's the job of the addon owners.

**Comment by stefansaraev** [comment]
we may stop shipping them, but a user may chose to install them all, that's not in our control. not shipping per default solves only part of the issue :)

**Comment by MilhouseVH** [comment]
> or is it time to stop shipping pre-installed pvr clients as opdenkamp suggested one year ago ?

Yes, of course, I can do that and then I have to host them somewhere and keep them all up to date and versioned on an almost daily basis. And when a user upgrades/downgrades their OE installation, I'll have no idea what PVR add-on they actually have installed in their system when reporting a problem (they may have a newer PVR add-on but older OE base, or vice versa, which might lead to a conflict).

From a testing POV, I either ship all PVR add-ons (so that I have control over and full knowledge of what the user has installed), or no PVR add-ons (and that will be the end of any bleeding-edge PVR testing).

Trust me, I regularly consider binning the PVR addons as they're nothing but a total PITA (but sadly that wouldn't stop being the case if I were to no longer ship them as part of the build - I'd then get reports of problems from users installing God-knows what version of the add-on, probably some months old version from the OE repo).

**Comment by stefansaraev** [comment]
@MilhouseVH I know and I understand you. "Ive been there..", that's why I moved everything to openelec repo long time ago.

**Comment by FernetMenta** [comment]
Why should a user install half a dozen pvr addons or even more? I think that is not a common use case we need to take care.
There is a real issue with audiodecoder addons because those take precedence over internal codecs when installed. This is no-go if a user did not explicitly enabled this feature.

**Comment by stefansaraev** [comment]
well, in past few years, I've seen users doing all sort of weird stuff, because they can. if it's allowed in the GUI - they do it. if it doesnt work - it is a bug ;)

**Comment by MilhouseVH** [comment]
Perhaps I have the wrong end of the stick, but I don't understand why we're having this problem now that we have `addons-manifest.xml`. We're now able to define "system" addons, but somehow we can't differentiate pre-installed add-ons from system add-ons, so pre-installed add-ons must be auto-enabled too? This is crazy.

If it became necessary, I'd happily add the pre-installed add-ons to `addons-manifest.xml` if there were a way to configure their default/initial state, ie. enabled or disabled.

**Comment by stefansaraev** [comment]
system addons != pre-installed addons. system are those defined in addon manifest. those are critical to kodi operation and should NOT be allowed to be disabled/uninstalled. everything else is "user" addons.

yes. ALL addons should be enabled. the point of this PR..

added a commit to workaround pvr clients / audio decoders. is it acceptable ?

**Comment by FernetMenta** [comment]
I agree with the general idea of this PR. Wouldn't it be better to have a manifest for pre-installed addons that allows controlling this? Could be black or whitelist, or both.

**Comment by stefansaraev** [comment]
you cant have a manifest for pre-installed addns. you can never know what pre-installed addons a user may have.

**Comment by MilhouseVH** [comment]
> you can never know what pre-installed addons a user may have.

Surely the user (or in my case, system builder) that is adding the pre-installed addons would know, and could adapt the "pre-installed manifest" accordingly?

**Comment by stefansaraev** [comment]
@MilhouseVH here my addons http://sprunge.us/gdBI
so, you are telling me that I have to adapt some xml file every time my addons*.db goes puff of blue smoke ? nope. and if I dont want to do that, neither a regular rpi user would.

**Comment by MilhouseVH** [comment]
No, I'm talking about "pre-installed" addons, those included or shipped as part of a build, not those installed by the user (using "install from zip", "install from repository" etc.).

Maybe I'm using incorrect terminology but we seem to have "system" addons on which Kodi depends, "pre-installed" addons (value-add or convenience addons added by the system builder), and "user installed" addons. The "system" addons are managed by `addons-manifest.xml` and always enabled, while "user installed" addons are always enabled regardless. This is all fine.

The problem is with the "pre-installed" add-ons that fall between the two stools - they're shipped as part of an installation (ie. not installed by the user) but some of the addons could/should be enabled by default (eg. inputstream.mpd), while others should not (pvr.*, audioencoder.*).

I don't see a problem having these "pre-installed" addons added to a manifest file so that the user/builder can dictate how the system initialises the add-on system with respect to any pre-installed add-ons.

**Comment by FernetMenta** [comment]
> so, you are telling me that I have to adapt some xml file every time my addons*.db goes puff of blue smoke ? nope

I see, this is a valid requirement but also an additional use case. This is not first start of Kodi but start after some db corruption.

**Comment by MilhouseVH** [comment]
By the way this PR (auto-disabling pvr.\* and audiodecoder.*) is working well, at least with my OE test builds. I've tested upgrading Addons21.db, and starting without any Addons*.db - system comes up normally in both scenarios.

File: xbmc/addons/AddonDatabase.cpp [hunk 1]
```diff
@@ -218,7 +218,7 @@ void CAddonDatabase::SyncInstalled(const std::set<std::string>& ids, const std::
     BeginTransaction();
     for (const auto& id : added)
       m_pDS->exec(PrepareSQL("INSERT INTO installed(addonID, enabled, installDate) "
-          "VALUES('%s', '%d', '%s')", id.c_str(), enabled.find(id) != enabled.end() ? 1 : 0, now.c_str()));
+          "VALUES('%s', 1, '%s')", id.c_str(), now.c_str()));
```
Comment (garbear) [inline xbmc/addons/AddonDatabase.cpp]: The `enabled` parameter added in 108db2206194651cc72883b344f9911707a386c3 looks to be unused now

**Comment by stefansaraev** [comment]
> I see, this is a valid requirement but also an additional use case. This is not first start of Kodi but start after some db corruption.

for addonmanager. this is first start. addonmanager does not care, and dont have to. why would addonmanager care about if musicdb or videodb or guisettings and whatever else exists ?

btw, we can't recover at all after *real* db corruption, on db corruption user will be forced to manualy delete addon db. same after failed/messed up db upgrade. happened countless of times for many users in the past. will happen in the future :)

**Comment by tamland** [comment]
As I explained in the other PR this was change was intentional precisely to avoid all the issues that have been mentioned here. It's fine to change it back again as it's purely a technical detail that has no affect for any normal use. If there's any issues, report it. If you want to change it, the least I expect is a non-hacky properly thought-out proposal for an alternative solution (I can even help you implement it).

But all I have heard so far is "should this" "should that", "this isn't how it used to be", "it broke my workflow", and ramblings about a bunch of unrelated issues.

**Comment by garbear** [comment]
Can PVR add-ons and audio decoders be special-cased? A whitelist or blacklist will fail the next time a PVR add-on or peripheral add-on is packaged with an unmodified build. Users are starting to complain of disabled joystick support on the forum.

**Comment by stefansaraev** [comment]
sure, it's what this PR was doing.

**Comment by garbear** [comment]
And this PR is too hacky, right? How can we do exactly the same with being less hacky?

**Pull request closed without merging**